src/scvi/external/mrvi/_module.py

```diff
@@ -0,0 +1,579 @@
+    """Multi-resolution Variational Inference (MrVI) module.
+
+    Parameters
+    ----------
+    n_input
+        Number of input features.
+    n_sample
+        Number of samples.
+    n_batch
+        Number of batches.
+    n_labels
+        Number of labels.
+    n_latent
+        Number of latent variables.
+    n_latent_u
+        Number of latent variables for ``u``.
+    encoder_n_hidden
+        Number of hidden units in the encoder.
+    encoder_n_layers
+        Number of layers in the encoder.
+    z_u_prior
+        Whether to place a Gaussian prior on ``z`` given ``u``.
+    z_u_prior_scale
+        Natural log of the scale parameter of the Gaussian prior placed on ``z`` given ``u``. Only
+        applies of ``learn_z_u_prior_scale`` is ``False``.
+    u_prior_scale
+        Natural log of the scale parameter of the Gaussian prior placed on ``u``. If
+        ``u_prior_mixture`` is ``True``, this scale applies to each mixture component distribution.
+    u_prior_mixture
+        Whether to use a mixture of Gaussians prior for ``u``.
+    u_prior_mixture_k
+        Number of mixture components to use for the mixture of Gaussians prior on ``u``.
+    learn_z_u_prior_scale
+        Whether to learn the scale parameter of the prior distribution of ``z`` given ``u``.
+    scale_observations
+        Whether to scale the loss associated with each observation by the total number of
+        observations linked to the associated sample.
+    px_kwargs
+        Keyword arguments for the generative model.
+    qz_kwargs
+        Keyword arguments for the inference model from ``u`` to ``z``.
+    qu_kwargs
+        Keyword arguments for the inference model from ``x`` to ``u``.
+    training
+        Whether the model is in training mode.
+    n_obs_per_sample
+        Number of observations per sample.
+    """
+
+    n_input: int
+    n_sample: int
+    n_batch: int
+    n_labels: int
+    n_latent: int = 30
+    n_latent_u: int = 10
+    encoder_n_hidden: int = 128
+    encoder_n_layers: int = 2
+    z_u_prior: bool = True
+    z_u_prior_scale: float = 0.0
+    u_prior_scale: float = 0.0
+    u_prior_mixture: bool = True
+    u_prior_mixture_k: int = 20
+    learn_z_u_prior_scale: bool = False
+    scale_observations: bool = False
+    px_kwargs: dict | None = None
+    qz_kwargs: dict | None = None
+    qu_kwargs: dict | None = None
+    training: bool = True
+    n_obs_per_sample: jax.typing.ArrayLike | None = None
+
+    def setup(self):
+        px_kwargs = DEFAULT_PX_KWARGS.copy()
+        if self.px_kwargs is not None:
+            px_kwargs.update(self.px_kwargs)
+
+        qz_kwargs = DEFAULT_QZ_ATTENTION_KWARGS.copy()
+        if self.qz_kwargs is not None:
+            qz_kwargs.update(self.qz_kwargs)
+
+        qu_kwargs = DEFAULT_QU_KWARGS.copy()
+        if self.qu_kwargs is not None:
+            qu_kwargs.update(self.qu_kwargs)
+
+        is_isomorphic_uz = self.n_latent == self.n_latent_u
```

Comment:
I'd like to raise a warning if n_latent is set lower than n_latent_u. what's the best way to do this @martinkim0 ?

Reply:
I can take a look at this!